### PR TITLE
Component/details

### LINKED ISSUE #24: As a user I should be able to click on a market and view more details about it at '/markets/:id'

### PR DESCRIPTION
**Pull request by Meekb**
### Acceptance Criteria
When a Results card is clicked, the url changes to /markets/:id and the page changes and displays all details for the clicked market

### Issues Resolved
Closes #24 

### Code Reviewers
@clairefields15 
@AshleyOh-bit 
@alexmkio

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -89,11 +89,14 @@ export const App: React.FC = () => {
...
           )}
         />
-        {/* <Route
+        <Route
           exact
           path='/markets/:id'
-          render={({ match }) => <Details id={parseInt(match.params.id)} />}
-        /> */}
+          render={({ match }) => {
+            const { id } = match.params
```
Comment (alexmkio) [inline src/App.tsx:96]: Interesting. I've only previously seen this sent in as props this way: `id={match.params.id}`, but if it works it works! Thank you benevolent JavaScript overlords.

File: src/components/Details/Details.tsx [hunk 2]
```diff
@@ -1,7 +1,67 @@
...
+  }[],
+  id: string;
+}
+
+const Details: React.FC<MarketProps> = ({ id, markets, marketDetails }) => {
+
+  const marketMatch = markets.filter(market => market.id === Number(id));
+  const detailsMatch = marketDetails.filter(market => { 
+    let match = market.mapsLink.split('%22')[1].includes(marketMatch[0].marketName.split(' ')[0]);
+    console.log(marketMatch[0].marketName)
```
Comment (alexmkio) [inline src/components/Details/Details.tsx]: This console.log could go

File: src/components/Details/Details.tsx [hunk 3]
```diff
@@ -1,7 +1,67 @@
...
+    return match
```
Comment (alexmkio) [inline src/components/Details/Details.tsx:31]: This _could_ be a direct return of line 30. No need to let variable "match", but it works!

File: src/components/Details/Details.tsx [hunk 4]
```diff
@@ -1,7 +1,67 @@
...
+  });
+  const nameMatch = marketMatch[0].marketName;
+  const detail = detailsMatch[0];
+  const openSeason = `Season: ${detail.schedule[0].season}`;
+  const daysAndTimes = `Open: ${detail.schedule[0].dayOfWeek} ${detail.schedule[0].time}`;
+  const productList = detail.products.map(prod => {
+    return (
+    <div className='list'>
+      <ul>
+        <li> {prod}, </li>
```
Comment (alexmkio) [inline src/components/Details/Details.tsx]: Is this going to render the last product with a trailing ","?

File: src/components/Details/Details.tsx [hunk 5]
```diff
@@ -1,7 +1,67 @@
...
+      </ul>
+    </div>
+    );
+  });
+
+  return (
+    <section className='market-details'>
+      <h1> {nameMatch} </h1>
```
Comment (alexmkio) [inline src/components/Details/Details.tsx:51]: H1 level is reserved for our application name (in the footer in the wireframe). There can only be one H1.

File: src/components/Details/Details.tsx [hunk 6]
```diff
@@ -1,7 +1,67 @@
...
+        <p> Located at: {detail.street}, {detail.city} {detail.state}, {detail.zip} </p>
+      <div className='schedule'>
+        <h4> Season and Schedule: </h4>
```
Comment (alexmkio) [inline src/components/Details/Details.tsx:54]: I think best practice is no whitespace between html elements. I.e.: `<h4>Season and Schedule:</h4>`
[Turing style guide](https://github.com/turingschool-examples/html)

File: src/components/Details/Details.tsx [hunk 7]
```diff
@@ -1,7 +1,67 @@
...
+        <p> {openSeason} </p>
+        <p> {daysAndTimes} </p>
+      </div>
+      <div className='prod-list'>
+        <h4> Products available at this market: </h4>
```
Comment (alexmkio) [inline src/components/Details/Details.tsx:59]: Is there an h2 or h3 on this page?

**Review by alexmkio — CHANGES_REQUESTED** [review]

File: src/components/Details/Details.tsx (hunk 2 above)
Comment (Meekb) [reply src/components/Details/Details.tsx]: tis gone

File: src/components/Details/Details.tsx (hunk 5 above)
Comment (Meekb) [reply src/components/Details/Details.tsx:51]: is this only regarding h1? changed.

File: src/components/Details/Details.tsx (hunk 6 above)
Comment (Meekb) [reply src/components/Details/Details.tsx]: wow! personally I find it's incredibly hard to read so I've done it for my own readability but haven't specifically been given feedback regarding that in the past - so I appreciate you bringing this to my attention. All fixed.

File: src/components/Details/Details.tsx (hunk 7 above)
Comment (Meekb) [reply src/components/Details/Details.tsx]: now h2 name
h3 season and prods

File: src/components/Details/Details.tsx (hunk 4 above)
Comment (Meekb) [reply src/components/Details/Details.tsx]: Comma removed. I have tried within the .map  {prod.indexOf(prod) === detail.products.length - 1 && <li>propWithoutComma<>} as well as a ternary.

File: src/App.tsx (hunk 1 above)
Comment (Meekb) [reply src/App.tsx:96]: yep - went off how Em and I completed the card clicks in RT

File: src/components/Details/Details.tsx (hunk 3 above)
Comment (Meekb) [reply src/components/Details/Details.tsx:31]: oh yeah this is changed too

File: src/components/Details/Details.tsx (hunk 5 above)
Comment (alexmkio) [reply src/components/Details/Details.tsx]: correct. You can have as many h2, h3, ... as you need but only one H1

**Review by alexmkio — APPROVED** [review]

File: src/components/Details/Details.tsx [hunk 8]
```diff
@@ -1,7 +1,66 @@
...
+    mapsLink: string
+  }[],
+  id: string;
+}
+
+export const Details: React.FC<MarketProps> = ({ id, markets, marketDetails }) => {
+
+  const marketMatch = markets.filter(market => market.id === Number(id));
+  const detailsMatch = marketDetails.filter(market => { 
+    const match = market.mapsLink.split('%22')[1].includes(marketMatch[0].marketName.split(' ')[0]);
```
Comment (clairefields15) [inline src/components/Details/Details.tsx:30]: Should we wait on merging this until Alex/someone can get the id out of our utils function?

**Review by clairefields15 — COMMENTED** [review]
Nice work! I just wonder if we should finish adding that id to the cleaning function and refactor this before merging?

**Review by clairefields15 — APPROVED** [review]

**Pull request merged**